lib/CCreatureSet.cpp

```diff
@@ -659,14 +659,33 @@ std::string CStackInstance::bonusToGraphics(const std::shared_ptr<Bonus>& bonus)
 	return VLC->getBth()->bonusToGraphics(bonus);
 }
 
-void CStackInstance::setArmyObj(const CArmedInstance *ArmyObj)
+void CStackInstance::setArmyObj(const CArmedInstance * ArmyObj)
 {
 	if(_armyObj)
 		detachFrom(const_cast<CArmedInstance*>(_armyObj));
 
 	_armyObj = ArmyObj;
 	if(ArmyObj)
 	{
+		if(type)
+		{
+			auto & creatureBonuses = const_cast<CCreature *>(type)->getExportedBonusList();

```

Comment:
Opposite Bonus from creature should not have propagator, corresponding opposite bonus should be exported from Stack Instance because creature bonuses are shared between all players and can't have owner.
How 'attachTo' is  related here - unclear.